quil-rs/src/program/graph.rs

```diff
@@ -541,17 +544,17 @@ fn terminate_working_block(
     Ok(())
 }
 
-impl ScheduledProgram {
+impl<'a> ScheduledProgram<'a> {
     /// Structure a sequential program
     #[allow(unused_assignments)]
-    pub fn from_program(program: &Program) -> ScheduleResult<Self> {
+    pub fn from_program(program: &'a Program) -> ScheduleResult<Self> {
         let mut working_label = None;
-        let mut working_instructions: Vec<Instruction> = vec![];
+        let mut working_instructions: Vec<&'a Instruction> = vec![];
         let mut blocks = IndexMap::new();
 
-        let instructions = program.to_instructions();
+        // let instructions = program.to_instructions();
 
-        for (index, instruction) in instructions.into_iter().enumerate() {
+        for (index, instruction) in program.instructions.iter().enumerate() {
```

Comment:
Per @kalzoo, the "headers" (everything added to the `Vec<Instruction>` returned by `to_instructions()` that is _not_ part of `program.instructions`) are not used here, so we can iterate over the `instructions` field directly.

Reply:
yep. We'll need to document this intent somewhere - all instructions within all blocks must be completely contained within `Program.instructions` (the "body" of the program) in order for this to be valid.

I think a (breaking) rename of `.instructions` to `.body` or `.instruction_body` might help make this clear, together with making it [crate-]private and then exposing a getter.